src/osd/OSDMap.cc

```diff
@@ -5799,7 +5800,7 @@ bool OSDMap::try_drop_remap_overfull(
     for (auto um_pair : pg_upmap_items) {
       auto& um_from = um_pair.first;
       auto& um_to = um_pair.second;
-      if (um_to == osd) {
+      if (um_to == osd && osd_deviation[um_from] + 1 < osd_deviation[osd]) {
```

Comment:
Better readability:
```suggestion
      if ((um_to == osd) && (osd_deviation[um_from] + 1 < osd_deviation[osd])) {
```

Reply:
> Better readability:

I have modified the code according to your suggestions and tested it. Please kindly help me review it again and if there is no problem, please help me merge it.

Reply:
This part looks good